bundles/org.openhab.core.model.rule/src/org/openhab/core/model/rule/jvmmodel/RulesRefresher.java

```diff
@@ -50,30 +51,65 @@
  * @author Maoliang Huang - refactor
  */
 @Component(immediate = true, service = {})
+@NonNullByDefault
 public class RulesRefresher implements ReadyTracker {
 
-    // delay in ms before rule resources are refreshed after items or services have changed
-    private static final long REFRESH_DELAY = 5000;
+    // delay in seconds before rule resources are refreshed after items or services have changed
+    private static final long REFRESH_DELAY = 5;
 
     private static final String POOL_NAME = "automation";
     public static final String RULES_REFRESH = "rules_refresh";
 
     private final Logger logger = LoggerFactory.getLogger(RulesRefresher.class);
 
-    private ScheduledFuture<?> job;
-    private ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(POOL_NAME);
+    private @Nullable ScheduledFuture<?> job;
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(POOL_NAME);
     private boolean started;
-    private ReadyMarker marker = new ReadyMarker("dsl", RULES_REFRESH);
-
-    private ItemRegistryChangeListener itemRegistryChangeListener;
-    private ThingRegistryChangeListener thingRegistryChangeListener;
+    private final ReadyMarker marker = new ReadyMarker("dsl", RULES_REFRESH);
 
     private final ModelRepository modelRepository;
     private final ItemRegistry itemRegistry;
     private final ThingRegistry thingRegistry;
     private final EventPublisher eventPublisher;
     private final ReadyService readyService;
 
+    private final ItemRegistryChangeListener itemRegistryChangeListener = new ItemRegistryChangeListener() {
```

Comment:
Oh, that's a worthwhile improvement - creating a new listener on every marker event was indeed a very bad idea. I'd bet that this change alone will make the RPi to only need 5 secs to be ready for rules now ;-)